core/eolearn/core/eodata_merge.py

```diff
@@ -121,15 +121,15 @@ def _return_if_equal_operation(arrays: np.ndarray) -> bool:
 
 def _merge_timestamps(
     eopatches: Sequence[EOPatch], reduce_timestamps: bool
-) -> tuple[list[dt.datetime], list[np.ndarray]]:
+) -> tuple[list[dt.datetime] | None, list[np.ndarray]]:
     """Merges together timestamps from EOPatches. It also prepares a list of masks, one for each EOPatch, how
     timestamps should be ordered and joined together.
     """
-    timestamps_per_eopatch = [eopatch.timestamps for eopatch in eopatches]
+    timestamps_per_eopatch = [eopatch.timestamps or [] for eopatch in eopatches]
```

Comment:
So, for the purpose of merging, if `eopatch.timestamps` is `None` it's temporarily converted to `[]`.

However, the final merged eopatch could still in principle have temporal features with empty temporal slices, in that case the `[]` timestamp should be eligible, right?

Reply:
you are correct... what a stupid edge case. Will add more firepower here. Great catch